pkg/endpoint/endpoint_test.go

```diff
@@ -343,7 +343,7 @@ func (s *EndpointSuite) TestWaitForPolicyRevision(c *C) {
 
 	e.state = StateCreating
 	ctx, cancel = context.WithCancel(context.Background())
-	ch = e.WaitForPolicyRevision(ctx, 99)
+	ch = e.WaitForPolicyRevision(ctx, 99, nil)
```

Comment:
Maybe we should add a test case for `WaitForPolicyRevision` with some non-nil `done` function?

Reply:
@mrostecki Good point :) I shouldn't have been lazy. I added a check in the same test, since the callback should be called whenever the channel is closed. Let me know what you think, I left comments but they aren't really that useful :/